1.6

Chris-Castle opened the pull request:
Bug Fix on delete and rules based redirect

thecadams commented:
Chris,
This is an amazing amount of work - thanks so much for contributing! Unfortunately I don't have a windows environment right now, so I'm wondering - can you point me in the right direction to test it? Would you recommend using an Azure VM and putting Sitecore on there, or something else?

thecadams commented on README.md:
```diff
@@ -6,6 +6,17 @@ An improved version of the Sitecore 301 redirect module. Original version was cr
 Improvements by [Chris Adams](http://www.cadams.com.au/), [Max Slabyak](https://github.com/maxslabyak) and [Mark Wiseman](https://github.com/mawiseman)
```
Should add your name here!

thecadams commented on data/packages/301RedirectModule-1.5.xml:
```diff
@@ -16,6 +16,19 @@ An improved version of the Sitecore 301 redirect module. Original version was cr
 Improvements by [Chris Adams](http://www.cadams.com.au/), [Max Slabyak](https://github.com/maxslabyak) and [Mark Wiseman](https://github.com/mawiseman)
```
And your name here!

thecadams commented on data/packages/301RedirectModule-1.6.xml:
```diff
@@ -1,202 +1,303 @@
+
+An improved version of the Sitecore 301 redirect module. Original version was created by Chris Castle, available at [http://trac.sitecore.net/301RedirectModule](http://trac.sitecore.net/301RedirectModule)
+
+Improvements by [Chris Adams](http://www.cadams.com.au/), [Max Slabyak](https://github.com/maxslabyak) and [Mark Wiseman](https://github.com/mawiseman)
```
And here

thecadams commented on source/Handlers/AutoCreateRedirectOnMove.cs:
```diff
@@ -0,0 +1,74 @@
+                if (parentItem != null)
+                {
+                    //Now we need to get the template from which the item is created (Redirect Url)
+                    TemplateItem template = db.GetTemplate(new ID("{B5967A68-7F70-42D3-9874-0E4D001DBC20}"));
```
Probably worth extracting this to a constant

thecadams commented on source/Handlers/AutoCreateRedirectOnMove.cs:
```diff
@@ -0,0 +1,74 @@
+                        {
+                            // Assign values to the fields of the new item
+                            newItem.Fields["Requested Url"].Value = oldPath;
+                            newItem.Fields["Response Status Code"].Value = "{3184B308-C050-4A16-9F82-D77190A28F0F}";  // 301
```
And this one

thecadams commented on source/Handlers/AutoCreateRedirectOnMove.cs:
```diff
@@ -0,0 +1,74 @@
+                string oldPath = LinkManager.GetItemUrl(oldParent).Replace("/sitecore/shell", "") + "/" + LinkManager.GetItemUrl(item).Split('/').Last();
+                Database db = Sitecore.Configuration.Factory.GetDatabase("master");
+                // Get the generated folder underneath the redirects folder.  It is a bucketed item.
+                Item parentItem = db.GetItem(new ID("{46CE2092-FF8D-454E-B826-A2ADDB7E0BA3}"));
```
Probably worth extracting this to a constant

Chris-Castle commented:
Hi Chris,

I don’t expect much impact because of the way the code works.  The module itself does nothing unless you ask for a page that Sitecore can’t resolve.  Then, it looks for a direct match, pattern match or rules match in that order.  So if you are primarily using direct matches, the new code rarely runs.  The only think I considered doing is using the index to look for a match instead of the Sitecore APIs against the content tree.

As for the other new code such as the event handler on item move, these are simple and have very little performance impact.

Thanks,
Chris

Chris Castle ● Sitecore  ● o:281-516-9861 ● m:415-385-1830 ● chc@sitecore.net<mailto:chc@sitecore.net>

From: Chris Adams [mailto:notifications@github.com]
Sent: Tuesday, October 11, 2016 12:26 AM
To: thecadams/301RedirectModule <301RedirectModule@noreply.github.com>
Cc: Chris Castle <chc@sitecore.net>; Author <author@noreply.github.com>
Subject: Re: [thecadams/301RedirectModule] 1.6 (#18)


@thecadams commented on this pull request.

This is great Chris!

I notice a lot more code is running each request - so wondering if you have any idea of the perf impact of this change?

________________________________

In README.md<https://github.com/thecadams/301RedirectModule/pull/18#pullrequestreview-3612626>:

> @@ -6,6 +6,17 @@ An improved version of the Sitecore 301 redirect module. Original version was cr

 Improvements by [Chris Adams](http://www.cadams.com.au/), [Max Slabyak](https://github.com/maxslabyak) and [Mark Wiseman](https://github.com/mawiseman)

Should add your name here!

________________________________

In data/packages/301RedirectModule-1.5.xml<https://github.com/thecadams/301RedirectModule/pull/18#pullrequestreview-3612626>:

> @@ -16,6 +16,19 @@ An improved version of the Sitecore 301 redirect module. Original version was cr

 Improvements by [Chris Adams](http://www.cadams.com.au/), [Max Slabyak](https://github.com/maxslabyak) and [Mark Wiseman](https://github.com/mawiseman)

And your name here!

________________________________

In data/packages/301RedirectModule-1.6.xml<https://github.com/thecadams/301RedirectModule/pull/18#pullrequestreview-3612626>:

> +<project>

+  <Metadata>

+    <metadata>

+      <PackageName>301 Redirect Module</PackageName>

+      <Author>Chris Adams, Max Slabyak, Mark Wiseman, and Chris Castle</Author>

+      <Version>1.6</Version>

+      <Revision />

+      <License />

+      <Comment>A modification of a package contributed to Sitecore Shared Source by Chris Castle. Original is available at http://trac.sitecore.net/301RedirectModule</Comment<http://trac.sitecore.net/301RedirectModule%3c/Comment>>

+      <Attributes />

+      <Readme>301RedirectModule

+=================

+

+An improved version of the Sitecore 301 redirect module. Original version was created by Chris Castle, available at [http://trac.sitecore.net/301RedirectModule](http://trac.sitecore.net/301RedirectModule)

+

+Improvements by [Chris Adams](http://www.cadams.com.au/), [Max Slabyak](https://github.com/maxslabyak) and [Mark Wiseman](https://github.com/mawiseman)

And here

________________________________

In source/Handlers/AutoCreateRedirectOnMove.cs<https://github.com/thecadams/301RedirectModule/pull/18#pullrequestreview-3612626>:

> +        }

+

+        private void CreateRedirectItem(Item item, Item oldParent)

+        {

+            // we only want a redirect on pages and media assets

+            if (oldParent.Paths.IsContentItem || oldParent.Paths.IsMediaItem)

+            {

+                string oldPath = LinkManager.GetItemUrl(oldParent).Replace("/sitecore/shell", "") + "/" + LinkManager.GetItemUrl(item).Split('/').Last();

+                Database db = Sitecore.Configuration.Factory.GetDatabase("master");

+                // Get the generated folder underneath the redirects folder.  It is a bucketed item.

+                Item parentItem = db.GetItem(new ID("{46CE2092-FF8D-454E-B826-A2ADDB7E0BA3}"));

+

+                if (parentItem != null)

+                {

+                    //Now we need to get the template from which the item is created (Redirect Url)

+                    TemplateItem template = db.GetTemplate(new ID("{B5967A68-7F70-42D3-9874-0E4D001DBC20}"));

Probably worth extracting this to a constant

________________________________

In source/Handlers/AutoCreateRedirectOnMove.cs<https://github.com/thecadams/301RedirectModule/pull/18#pullrequestreview-3612626>:

> +                if (parentItem != null)

+                {

+                    //Now we need to get the template from which the item is created (Redirect Url)

+                    TemplateItem template = db.GetTemplate(new ID("{B5967A68-7F70-42D3-9874-0E4D001DBC20}"));

+

+                    if (template != null)

+                    {

+                        // Create the item

+                        Item newItem = parentItem.Add(ItemUtil.ProposeValidItemName(oldPath.Replace('/',' ')), template);

+

+                        newItem.Editing.BeginEdit();

+                        try

+                        {

+                            // Assign values to the fields of the new item

+                            newItem.Fields["Requested Url"].Value = oldPath;

+                            newItem.Fields["Response Status Code"].Value = "{3184B308-C050-4A16-9F82-D77190A28F0F}";  // 301

And this one

________________________________

In source/Handlers/AutoCreateRedirectOnMove.cs<https://github.com/thecadams/301RedirectModule/pull/18#pullrequestreview-3612626>:

> +                using (new SecurityDisabler())

+                {

+                    CreateRedirectItem(item, item.Database.GetItem(oldParentID));

+                }

+            }

+        }

+

+        private void CreateRedirectItem(Item item, Item oldParent)

+        {

+            // we only want a redirect on pages and media assets

+            if (oldParent.Paths.IsContentItem || oldParent.Paths.IsMediaItem)

+            {

+                string oldPath = LinkManager.GetItemUrl(oldParent).Replace("/sitecore/shell", "") + "/" + LinkManager.GetItemUrl(item).Split('/').Last();

+                Database db = Sitecore.Configuration.Factory.GetDatabase("master");

+                // Get the generated folder underneath the redirects folder.  It is a bucketed item.

+                Item parentItem = db.GetItem(new ID("{46CE2092-FF8D-454E-B826-A2ADDB7E0BA3}"));

Probably worth extracting this to a constant

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/thecadams/301RedirectModule/pull/18#pullrequestreview-3612626>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AJ9EA3pOB-P8oa1kpCcH_rvU_hXPn91Wks5qyx38gaJpZM4KQnOe>.

thecadams commented:
Looks good then, merging. Congrats and thanks again! :beer:
